src/main/java/org/opensearch/flowframework/model/WorkflowNode.java

```diff
@@ -141,7 +150,11 @@ public static WorkflowNode parse(XContentParser parser) throws IOException {
                                 userInputs.put(inputFieldName, parser.text());
                                 break;
                             case START_OBJECT:
-                                userInputs.put(inputFieldName, parseStringToStringMap(parser));
+                                if (LLM_FIELD.equals(inputFieldName)) {
```

Comment:
Im concerned this handling of specific fields may not scale well as we add more and more steps.

I see that the `llm` field just consists of a `model_id` string and `parameters` that we can turn into a string to string map. Can we not instead just have the user provide these fields instead of the entire `llm` object and just build the `llm` object within this `RegisterAgentStep`? They'll of course be optional fields since not all agents require an llm.

Reply:
Yes. This is one of the TODO item. I can handle it in the follow up PR.
Also, `llm` object is required for the agent. `modelId`is optional for `Tools`.

Reply:
`llm` is only required for the chain of thought agent though right? but not the math agent, as per the documentation, the math agent would only require the following fields  : 
```
"name": "Test_Agent_For_Math",
    "type": "flow",
    "description": "this is a test agent",
    "tools": [<array of tools>]
```

Reply:
Discussed with @joshpalis. We can flattened `llm` fields to avoid special case handling. I'll take this up in the follow up PR.